astropy/units/tests/test_equivalencies.py

```diff
@@ -144,6 +144,20 @@ def test_bad_restfreqs(function, value):
         function(value)
 
 
+@pytest.mark.parametrize(('z', 'rv_ans'),
+                         [(0, 0 * (u.km / u.s)),
+                          (0.001, 299642.56184583 * (u.m / u.s)),
+                          (-1, -2.99792458e8 * (u.m / u.s))])
+def test_doppler_redshift(z, rv_ans):
+    from astropy.cosmology.units import redshift
+    for z_unit in (u.dimensionless_unscaled, redshift):
+        z_in = z * z_unit
+        rv_out = z_in.to(u.km / u.s, u.doppler_redshift())
+        z_out = rv_out.to(z_unit, u.doppler_redshift())
```

Comment:
But this doesn't have to. It's interpreting a doppler velocity as a redshift velocity, which is fine.
OTOH, one might want the equivalency to error, for simplicity.

Reply:
I agree. OK one way but not the other way is too confusing.